app/routes/bekk/routes.py

```diff
@@ -0,0 +1,177 @@
+
+        if not add:
+            delete = form["delete"].data
+            bekk = Bekk.query.filter_by(id=form["id"].data).one()
+
+            if delete:
+                db.session.delete(bekk)
+                db.session.commit()
+
+                shutil.rmtree(os.path.join(bekk_path, form["promo"].data, form["bekk_name"].data))
+
+                flask.flash(_("Bekk supprimé."))
+                return flask.redirect(flask.url_for("bekk.main", promo=promo))
+
+            else:
+                old_name = bekk.name
+                old_promo = str(bekk.promo)
+
+                bekk.name = form["bekk_name"].data
+                bekk.promo = form["promo"].data
+                bekk.date = form["date"].data
+
+                if old_name != bekk.name or old_promo != bekk.promo:
+                    old_path = os.path.join(bekk_path, old_promo, old_name)
+                    files = os.listdir(os.path.join(bekk_path, old_promo, old_name))
+                    for file in files:
+                        if file.split(".")[-1] == "png":
+                            os.rename(
+                                os.path.join(old_path, file),
+                                os.path.join(old_path, bekk.name + "_" + bekk.promo + "_" + file.split("_")[-1]),
+                            )
+                        else:
+                            os.rename(
+                                os.path.join(old_path, file),
+                                os.path.join(old_path, bekk.name + "_" + bekk.promo + ".pdf"),
+                            )
+
+                    if old_name != bekk.name:
+                        os.rename(old_path, os.path.join(bekk_path, old_promo, bekk.name))
+
+                    if old_promo != bekk.promo:
+                        if not os.path.exists(os.path.join(bekk_path, bekk.promo)):
+                            os.mkdir(os.path.join(bekk_path, bekk.promo))
+                        shutil.move(os.path.join(bekk_path, old_promo, bekk.name), os.path.join(bekk_path, bekk.promo))
+
+                db.session.commit()
+                flask.flash(_("Bekk édité."))
+                return flask.redirect(flask.url_for("bekk.main", promo=promo))
+
+    return flask.render_template(
+        "bekk/main.html",
+        title=_("Page du Bekk ESPCI"),
+        bekks=bekks,
+        view_promo=promo,
+        promos=promos,
+        bekk_path=bekk_path,
+        form=form,
+    )
+
+
+@bp.route("/reader/<int:id>", methods=["GET"])
+# @context.permission_only(PermissionType.read, PermissionScope.bekk)
+def reader(id: int) -> typing.RouteReturn:
+    """Bekk module main page"""
+
+    bekk = Bekk.query.filter_by(id=id).one_or_none()
+
+    filepath = os.path.join(
+        flask.current_app.config["BEKKS_BASE_PATH"],
+        str(bekk.promo),
+        bekk.name,
+    )
+    pages = range(len(os.listdir(filepath)) - 1)
+
+    return flask.render_template("bekk/reader.html", title=_(bekk.name), bekk=bekk, pages=pages)
```

Comment:
Ça ne sert à rien d'essayer de traduire un nom dynamique (stocké en base) :wink:

```suggestion
    return flask.render_template("bekk/reader.html", title=bekk.name, bekk=bekk, pages=pages)
```